Code/Framework/AzCore/AzCore/std/ranges/ranges_algorithm.h

```diff
@@ -992,5 +992,69 @@ namespace AZStd::ranges
     {
         constexpr Internal::find_end_fn find_end{};
     }
-} // namespace AZStd::ranges
 
+    namespace Internal
+    {
+        struct all_of_fn
+        {
+            template<class I, class S, class Proj = identity, class Pred,
```

Comment:
You need to swap the `Pred` and `Proj` parameters around, as Pred doesn't have a default value, while Proj does.
This will not compile on clang.